llvm/lib/Target/NVPTX/NVPTXISelDAGToDAG.cpp

```diff
@@ -3914,8 +3914,13 @@ bool NVPTXDAGToDAGISel::SelectADDRri_imp(
         Base = CurDAG->getTargetFrameIndex(FIN->getIndex(), mvt);
       else
         Base = Addr.getOperand(0);
-      Offset = CurDAG->getTargetConstant(CN->getZExtValue(), SDLoc(OpNode),
-                                         mvt);
+
+      // Offset is a signed i32 constant in PTX [register+offset] address mode
```

Comment:
Nit: Offset must fit in a 32-bit signed int.

Functionally, the code is fine, but I wonder if using `getSignificantBits() <= 32` instead would reflect your intent a bit better. I find the **is** part of `isSignedIntN(32)` to be somewhat misleading in this context. Naming it `fitsSignedIntN(32)` would be better, IMO. Maybe it's just me.

Reply:
I've update the comments. I agree that the name of this function should probably be `fitsSignedIntN` since that is what it's checking, regardless of context. That being said, I'd prefer to use the more specific function even it it may be a little misnamed.